ui/modelUtils.go

```diff
@@ -108,28 +100,28 @@ func (m *Model) executeKeybinding(key string) tea.Cmd {
 }
 
 func (m *Model) runCustomPRCommand(commandTemplate string, prData *data.PullRequestData) tea.Cmd {
-	repoName := prData.GetRepoNameWithOwner()
-	repoPath, ok := common.GetRepoLocalPath(repoName, m.ctx.Config.RepoPaths)
-
-	if !ok {
-		return func() tea.Msg {
-			return constants.ErrMsg{Err: fmt.Errorf("Failed to find local path for repo %s", repoName)}
-		}
+	// A generic map is a pretty easy & flexible way to populate a template if there's no pressing need
+	// for sructured data, existing structs, etc. Especially if holes in the data are expected.
+	input := map[string]any {
+		"RepoName":    prData.GetRepoNameWithOwner(),
+		"PrNumber":    prData.Number,
+		"HeadRefName": prData.HeadRefName,
+		"BaseRefName": prData.BaseRefName,
 	}
 
-	input := PRCommandTemplateInput{
-		RepoName:    repoName,
-		RepoPath:    repoPath,
-		PrNumber:    prData.Number,
-		HeadRefName: prData.HeadRefName,
-		BaseRefName: prData.BaseRefName,
+	// Append in the local RepoPath only if it can be found
+	if repoPath, ok := common.GetRepoLocalPath(input["RepoName"].(string), m.ctx.Config.RepoPaths); ok {
+		input["RepoPath"] = repoPath
 	}
 
 	cmd, err := template.New("keybinding_command").Parse(commandTemplate)
 	if err != nil {
 		log.Fatal("Failed parse keybinding template", err)
 	}
 
+	// Set the command to error out if required input (e.g. RepoPath) is missing
+	cmd = cmd.Option("missingkey=error")
```

Comment:
Hm, no I don't think that is the right approach...

Templates can be pretty complex. They can be recursively nested into a tree structure, be populated out in multiple stages by different calls to the `Execute` functions with different sources and types of data, potentially across multiple goroutines simultaneously.

The values passed in can be chained on one another (I presume to key into a map or struct which is another value passed in). They can be (zero-argument*) functions/methods that implements some kind of literal piping with `|` to produce the final substitution. I believe they may return a 2nd err return which will be propagated out ultimately.  <sup>( * there is a manner in which the final piece of data passed in, can have arguments, but honestly there's a lot going on in there, and a lot of ways and reasons it might spit errors at you. )</sub>

There's definitely cases where it will error on 'empty or incomplete' template, other internal/syntactic issues, unexpected data types or values, etc...

The `Option`  that this branch sets is essentially stating;

>   'This is our final pass over this Template, and we will consider any dangling placeholders left unfilled as   _also_  being errors.

While trivial current uses seem in the source to suggest the other values passed in are reliable, future additions may not be. The writer of the template might typo key/member names, screw up their syntax, etc. and cause an unrelated error, or the template may have its own reasons to.

It could certainly complicate the error handing if you want to tease any nuance out, though I think you may be able to simple propagate the errors, or wrapped versions of them, up to wherever and how that gets communicated to the user.

Reply:
I'm not sure how this relates to the fact you can show a log to the user using the `tea.Command` instead of crashing the app with fatal.Log.

```go
return func() tea.Msg { return constants.ErrMsg{Err: fmt.Errorf("Failed to parsetemplate %s", template)} }
```

Reply:
Sorry, think I totally misread/missed the point there... I didn't put any of the `log.Fatal()`'s in, so I think my mind went elsewhere.

There's a few sprinkled about, though I confess I'm not really up to speed with bubbletea yet, so not sure I was even aware what that snippet was doing tbh. >_>

Reply:
I made that change, though #362 would supersede this PR if it is the design direction you favour.